packages/nouns-contracts/test/foundry/NounsAuctionHouseV2.t.sol

```diff
@@ -0,0 +1,382 @@
+
+        vm.expectRevert(abi.encodeWithSelector(INounsAuctionHouse.AuctionHasntBegun.selector));
+        auction.settleAuction();
+    }
+
+    function test_V2Migration_works() public {
+        (NounsAuctionHouseProxy auctionProxy, NounsAuctionHouseProxyAdmin proxyAdmin) = _deployAuctionHouseV1AndToken(
+            owner,
+            noundersDAO,
+            minter
+        );
+        NounsAuctionHouse auctionV1 = NounsAuctionHouse(address(auctionProxy));
+        vm.prank(owner);
+        auctionV1.unpause();
+        vm.roll(block.number + 1);
+        (uint256 nounId, , uint256 startTime, uint256 endTime, , ) = auctionV1.auction();
+
+        address payable bidder = payable(address(0x142));
+        uint256 amount = 142 ether;
+        vm.deal(bidder, amount);
+        vm.prank(bidder);
+        auctionV1.createBid{ value: amount }(nounId);
+
+        _upgradeAuctionHouse(owner, proxyAdmin, auctionProxy);
+
+        NounsAuctionHouseV2 auctionV2 = NounsAuctionHouseV2(address(auctionProxy));
+        (
+            uint128 nounIdV2,
+            uint128 amountV2,
+            uint40 startTimeV2,
+            uint40 endTimeV2,
+            address payable bidderV2,
+            bool settledV2
+        ) = auctionV2.auction();
+
+        assertEq(nounIdV2, nounId);
+        assertEq(amountV2, amount);
+        assertEq(startTimeV2, startTime);
+        assertEq(endTimeV2, endTime);
+        assertEq(bidderV2, bidder);
+        assertEq(settledV2, false);
+    }
+
+    function test_auctionGetter_compatibleWithV1() public {
+        address bidder = address(0x4242);
+        vm.deal(bidder, 1.1 ether);
+        vm.prank(bidder);
+        auction.createBid{ value: 1.1 ether }(1);
+
+        NounsAuctionHouse auctionV1 = NounsAuctionHouse(address(auction));
+
+        (
+            uint128 nounIdV2,
+            uint128 amountV2,
+            uint40 startTimeV2,
+            uint40 endTimeV2,
+            address payable bidderV2,
+            bool settledV2
+        ) = auction.auction();
+
+        (
+            uint256 nounIdV1,
+            uint256 amountV1,
+            uint256 startTimeV1,
+            uint256 endTimeV1,
+            address payable bidderV1,
+            bool settledV1
+        ) = auctionV1.auction();
+
+        assertEq(nounIdV2, nounIdV1);
+        assertEq(amountV2, amountV1);
+        assertEq(startTimeV2, startTimeV1);
+        assertEq(endTimeV2, endTimeV1);
+        assertEq(bidderV2, bidderV1);
+        assertEq(settledV2, settledV1);
+    }
+
+    function bidAndWinCurrentAuction(address bidder, uint256 bid) internal returns (uint256) {
+        (uint256 nounId, , , uint256 endTime, , ) = auction.auction();
+        vm.deal(bidder, bid);
+        vm.prank(bidder);
+        auction.createBid{ value: bid }(nounId);
+        vm.warp(endTime);
+        auction.settleCurrentAndCreateNewAuction();
+        return block.timestamp;
+    }
+}
+
+contract NounsAuctionHouseV2_OracleTest is NounsAuctionHouseV2Test {
```

Comment:
@eladmallel by inheriting from `NounsAuctionHouseV2Test` it's also running all the tests again.
better to inherit from a contract without test methods

Reply:
good catch, fixing now